xarray/tests/test_indexing.py

```diff
@@ -144,6 +144,21 @@ def test_indexer(data, x, expected_pos, expected_idx=None):
                      [True, True, True, True, False, False, False, False],
                      pd.MultiIndex.from_product([[1, 2], [-1, -2]]))
 
+    def test_read_only_view(self):
+        from collections import OrderedDict
+        arr = DataArray(np.random.rand(3, 3),
+                        coords={'x': np.arange(3), 'y': np.arange(3)},
+                        dims=('x', 'y'))     # Create a 2D DataArray
+
+        arr = arr.expand_dims(OrderedDict([('z', 3)]), -1)  # New dimension 'z'
+        arr['z'] = np.arange(3)              # New coords to dimension 'z'
+        try:
+            mess = "No exception?!"
+            arr.loc[0, 0, 0] = 999
```

Comment:
`pytest.raises` is a nice way to  handle this boilerplate:
```
with pytest.raises(ValueError, match='Do you want to .copy()'):
    arr.loc[0, 0, 0] = 999
```

Reply:
Thank you. That is better than mine. Will change.